src/Interpreters/QueryNormalizer.cpp

```diff
@@ -134,6 +139,7 @@ void QueryNormalizer::visit(ASTIdentifier & node, ASTPtr & ast, Data & data)
             /// Check size of the alias before cloning too large alias AST
             alias_node->checkSize(data.settings.max_expanded_ast_elements);
             auto alias_name = ast->getAliasOrColumnName();
+            current_asts.insert(alias_node.get());
```

Comment:
This breaks some behaviour. Let's have only `checkStackSize();`. We will have TOO_DEEP_RECUSION instead of CYCLIC_ALIASES, but it's better that crash.
Also it's easier to backport.
I'll try to find proper fix later (of maybe if it's be too complicated it'll be better to leave with simple check of stack size and spend time on new analyzer).